sdcm/cluster.py

```diff
@@ -1708,10 +1710,32 @@ def get_ldap_config():
                 'ldap_bind_dn': f'cn=admin,{LDAP_BASE_OBJECT}',
                 'ldap_bind_passwd': LDAP_PASSWORD}
 
+    @staticmethod
+    def get_saslauthd_config():
+        if Setup.LDAP_ADDRESS is None:
+            return {}
+        ldap_server_ip = '127.0.0.1' if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION else Setup.LDAP_ADDRESS[0]
+        ldap_port = LDAP_SSH_TUNNEL_LOCAL_PORT if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION else \
+            Setup.LDAP_ADDRESS[1]
+        return {'ldap_servers': f'ldap://{ldap_server_ip}:{ldap_port}/',
+                'ldap_search_base': f'ou=Person,{LDAP_BASE_OBJECT}',
+                'ldap_bind_dn': f'cn=admin,{LDAP_BASE_OBJECT}',
+                'ldap_bind_pw': LDAP_PASSWORD}
+
     def create_ldap_users_on_scylla(self):
         self.run_cqlsh(f'CREATE ROLE \'{LDAP_ROLE}\' WITH SUPERUSER=true')
         for user in LDAP_USERS:
-            self.run_cqlsh(f'CREATE ROLE \'{user}\' WITH login=true AND password=\'{LDAP_PASSWORD}\'')
+            # Cannot create passwords with SaslauthdAuthenticator
+            self.run_cqlsh(f'CREATE ROLE \'{user}\' WITH login=true')
+        result = self.remoter.run("grep -o '^authenticator: .*' /etc/scylla/scylla.yaml")
+        if 'com.scylladb.auth.SaslauthdAuthenticator' in result.stdout:
+            opposite_auth = 'PasswordAuthenticator'
+            update_authenticator([self], opposite_auth)
+        # First LDAP_USERS will be used to alter system tables, so change it to superuser.
+        self.run_cqlsh(f'ALTER ROLE \'{LDAP_USERS[0]}\' with SUPERUSER=true and password=\'{LDAP_PASSWORD}\'')
```

Comment:
If saslauthdAuthenticator is used, the ldap users also need to be existed in scylla system_auth.
we have to create ldap users in scylla by 'cassandra' user.

For solving the chicken-egg problem, we need to create 'cassandra' in ldap server. login cqlsh by 'cassandra' user, create ldap users.

If saslauthdAuthenticator is used, the 'cassandra' can only add user(role) `without password`.
So the password is added by ALTER after switched to PasswordAuthenticator.
  we need to create 'cassandra' user in ldap server, then it can be used to login.

Reply:
> why not to use `cassandra \ cassandra` for it?

I used the same & default cassandra, and Roy suggested to use different account. Then we can verify that after switched the Authenticator, the new connection switches to a different account, and it works well.

> why not, not include the password in the previous cql command (`CREATE ROLE \'{LDAP_ROLE}\' WITH SUPERUSER=true`) if `SaslauthdAuthenticator` is enabled?